### PR TITLE
P5 8/menu page

### PR DESCRIPTION
**Pull request by ike-tom**
Hardcoded single menu item, will be synchronized with BE in the future. I will add redux after integration with BE.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { ReactElement, useState } from 'react';
 import { useQuery } from 'react-query';
 
 import helloRequest from './api/hello';
+import MenuPage from './components/UI/pages/MenuPage';
```
Comment (ruljin) [inline src/App.tsx]: I would recommend using `named exports`: look at Mateusz PR

File: src/App.tsx [hunk 2]
```diff
@@ -20,38 +21,9 @@ function ServerMessage() {
...
-            className="App-link"
-            href="https://vitejs.dev/guide/features.html"
-            target="_blank"
-            rel="noopener noreferrer"
-          >
-            Vite Docs
-          </a>
-        </p>
-      </header>
+      <MenuPage />
```
Comment (ruljin) [inline src/App.tsx]: You will have a conflict with Mateusz. Depends on which one PR will be merged first - it sounds like a lack of communication in a team

File: src/components/UI/organisms/MenuItem/MenuItem.css [hunk 3]
```diff
@@ -0,0 +1,114 @@
+@import url('https://fonts.googleapis.com/css2?family=Roboto:wght@400;700&display=swap');
+
+:root {
```
Comment (ruljin) [inline src/components/UI/organisms/MenuItem/MenuItem.css]: That root shouldn't be in this file, besides Mateusz is using `css modules` why you don't?

File: src/components/UI/organisms/MenuItem/MenuItem.css [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  --main: #fad390;
+  --secondary: #636e72;
+}
+
+.MenuItem {
```
Comment (ruljin) [inline src/components/UI/organisms/MenuItem/MenuItem.css]: You are all aware of the fact you are using `sass` that allows `inside` nesting? Thanks to that you can `simulate` BEM `classes` with inside `nestings` max to fourth level depth 😄

File: src/components/UI/organisms/MenuItem/MenuItem.css [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+  display: grid;
+  grid-template-columns: repeat(3, 1fr);
+  grid-template-rows: 1fr repeat(2, 0.5fr) repeat(2, 1fr);
+  grid-column-gap: 0px;
+  grid-row-gap: 0px;
+  background-color: var(--main);
+  align-items: center;
+  margin: 1rem;
```
Comment (ruljin) [inline src/components/UI/organisms/MenuItem/MenuItem.css]: NOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO decides in the project overall are you using `px` or `rem` - using them in parallel makes it in one point fight with each other about priority. Only `one` main format for the whole project - not less, not more.

File: src/components/UI/organisms/MenuItem/MenuItem.css [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+:root {
+  --main: #fad390;
+  --secondary: #636e72;
+}
+
+.MenuItem {
+  display: grid;
+  grid-template-columns: repeat(3, 1fr);
+  grid-template-rows: 1fr repeat(2, 0.5fr) repeat(2, 1fr);
+  grid-column-gap: 0px;
```
Comment (ruljin) [inline src/components/UI/organisms/MenuItem/MenuItem.css]: If you won't pass any value ... so not using this rule, won't it change anything?

File: src/components/UI/organisms/MenuItem/MenuItem.css [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+  grid-row-gap: 0px;
```
Comment (ruljin) [inline src/components/UI/organisms/MenuItem/MenuItem.css]: ☝️

File: src/components/UI/organisms/MenuItem/MenuItem.css [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+  background-color: var(--main);
+  align-items: center;
+  margin: 1rem;
+  padding: 1rem;
+  border: 2px solid var(--secondary);
+  border-radius: 15px;
+}
+
+.MuiTypography-root.menuItem__header {
```
Comment (ruljin) [inline src/components/UI/organisms/MenuItem/MenuItem.css]: If you would use `css modules` the `menu` prefix will be `redundant`

File: src/components/UI/organisms/MenuItem/MenuItem.css [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+}
+.menuItem__increase-button {
+  grid-area: 4 / 3 / 5 / 4;
+}
+.MuiButton-root.menuItem__more-info-button {
+  grid-area: 5 / 1 / 6 / 4;
+  margin-top: 1rem;
+}
+
+.menuItem__modal {
```
Comment (ruljin) [inline src/components/UI/organisms/MenuItem/MenuItem.css]: I said it in other `PR`

https://9elements.com/css-rule-order/

File: src/components/UI/organisms/MenuItem/MenuItem.css [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+  padding: 40px;
+  border: 2px solid var(--secondary);
+  box-shadow: 24;
+  width: 400;
+  background-color: var(--main);
+}
+
+.MuiTypography-root.menuItem__modal--additional {
+  font-size: 2rem;
+  font-weight: 700;
```
Comment (ruljin) [inline src/components/UI/organisms/MenuItem/MenuItem.css]: A good practice is to move the `repeatable` values of some styles to a `variables.scss` file which you `include` or `use` in other to use them - thanks to that if you change `one` variable for all elements in one place, you change them in the whole app

File: src/components/UI/organisms/MenuItem/MenuItem.tsx [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+import { Box, Button, Modal, TextField, Typography } from '@mui/material';
+import { useState } from 'react';
+
+function MenuItem() {
+  const [amount, setAmount] = useState(0);
+  const [open, setOpen] = useState(false);
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
+
+  const inc = () => {
```
Comment (ruljin) [inline src/components/UI/organisms/MenuItem/MenuItem.tsx]: This method name doesn't bring any knowledge about its usage. So ... what it is for?

File: src/components/UI/organisms/MenuItem/MenuItem.tsx [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+    setAmount((prevState) => prevState + 1);
+  };
+  const dec = () => {
```
Comment (ruljin) [inline src/components/UI/organisms/MenuItem/MenuItem.tsx]: ☝️

File: src/components/UI/organisms/MenuItem/MenuItem.tsx [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+      </Button>
+      <Button onClick={handleOpen} className="menuItem__more-info-button" variant="contained">
+        ADD INFO
+      </Button>
+      <Modal open={open} onClose={handleClose}>
+        <Box className="menuItem__modal">
+          <Typography className="menuItem__modal--additional">Additional Information</Typography>
+          <TextField multiline fullWidth minRows={5} placeholder="Additional Information" />
+          <Button fullWidth onClick={handleClose} className="menuItem__additional-i-button" variant="contained">
+            SAVE AND CLOSE
```
Comment (ruljin) [inline src/components/UI/organisms/MenuItem/MenuItem.tsx]: CSS `text-transform: uppercase`

File: src/components/UI/organisms/MenuItem/MenuItem.tsx [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+        +
+      </Button>
+      <Button onClick={handleOpen} className="menuItem__more-info-button" variant="contained">
+        ADD INFO
+      </Button>
+      <Modal open={open} onClose={handleClose}>
+        <Box className="menuItem__modal">
+          <Typography className="menuItem__modal--additional">Additional Information</Typography>
+          <TextField multiline fullWidth minRows={5} placeholder="Additional Information" />
+          <Button fullWidth onClick={handleClose} className="menuItem__additional-i-button" variant="contained">
```
Comment (ruljin) [inline src/components/UI/organisms/MenuItem/MenuItem.tsx]: This `className` has no sense

File: src/components/UI/pages/MenuPage.tsx [hunk 15]
```diff
@@ -0,0 +1,20 @@
+import './MenuPage.css';
```
Comment (ruljin) [inline src/components/UI/pages/MenuPage.tsx]: Modules `plox`

File: src/components/UI/pages/MenuPage.tsx [hunk 16]
```diff
@@ -0,0 +1,20 @@
...
+
+import MenuItem from '../organisms/MenuItem/MenuItem';
+
+function MenuPage() {
+  return (
+    <Box className="menu-page">
+      <Button className="menu-page__order-button" variant="contained">
+        ORDER
+      </Button>
+      <MenuItem />
```
Comment (ruljin) [inline src/components/UI/pages/MenuPage.tsx]: For showing up it's okay but overall you know you could just use:

`[...Array(n)].map((elementInArray, index) => ( 
    <div className="" key={i}>RANDOM</div> 
    ) 
)`

**Review by ruljin — CHANGES_REQUESTED** [review]
Request changes

File: src/components/UI/pages/MenuPage.tsx (hunk 16 above)
Comment (ike-tom) [reply src/components/UI/pages/MenuPage.tsx]: I wanted to have only one ORDER button at the top - not for every meal.

Comment (ruljin) [reply src/components/UI/pages/MenuPage.tsx]: This comment is for `MenuItem` not `Button`  😄

**Pull request merged**